entc/gen/type.go

```diff
@@ -708,16 +708,33 @@ func (t *Type) setupFKs() error {
 				}
 			}
 		}
-		// Special case for checking if the FK is already defined as the ID field (see issue 1288).
-		if key, _ := e.StorageKey(); key != nil && len(key.Columns) == 1 && key.Columns[0] == refid.StorageKey() {
-			fk.Field = refid
-			fk.UserDefined = true
+		// Special case for checking if the FK is already defined as the defined field (see issue 1288, 2205).
```

Comment:
Please, add a test case to help understand what this change does. Thanks

Reply:
Hi @a8m, i added a test case for this. Hope it helps 😄

Reply:
Hey @a8m, i added a test case for this. Hope it helps 😄

Reply:
Thanks, but integration tests are also necessary (see `entc/integration`).

Reply:
added @a8m. 😄